File: java/src/test/java/com/qmates/kata/extractandoverridecall/TestableOrderConfirmationService.java
package com.qmates.kata.extractandoverridecall;

public class TestableOrderConfirmationService extends OrderConfirmationService {

  @Override
  protected void sendEmail(String recipientEmailAddress, String subject, String body) {
    System.out.printf("Fake send mail subject [%s] to email address [%s]: \n%s", subject, recipientEmailAddress, body);
  }

}

File: typescript/src/04-extract-and-override-call/testable-order-confirmation.service.ts
import {OrderConfirmationService} from "./order-confirmation-service";

export class TestableOrderConfirmationService extends OrderConfirmationService {

  protected override async dispatchEmail(recipientEmail: string, subject: string, body: string): Promise<void> {
    console.log(`Fake send mail subject [${subject}] to email address [${recipientEmail}]: \n${body}`);
  }

}
